### DISCUSSION TITLE
`rosdep install` fails with `Cannot locate rosdep definitions` on Jetson Xavier

### DISCUSSION THREAD

**Discussion by raysu1027** [iRobotEducation/create3_docs#376] [Questions] [answered]
Hi 

I am step by step setup create3 but command **rosdep install --from-path src --ignore-src -yi** have some issues.

- Use environment

![image](https://user-images.githubusercontent.com/110970753/231964649-a353558f-fa9d-4cef-beac-a834054316d8.png)

- Error:

![image](https://user-images.githubusercontent.com/110970753/231936966-a6fea1ef-871e-44f8-87a2-c319f7e808b8.png)

ros2@ros2-desktop:~/create3_examples_ws$ rosdep install --from-path src --ignore-src -yi
ERROR: the following packages/stacks could not have their rosdep keys resolved
to system dependencies:
create3_teleop: Cannot locate rosdep definition for [teleop_twist_joy]
create3_lidar: Cannot locate rosdep definition for [slam_toolbox]

what can i do

**Comment by alsora** [comment]
Hi,
what version of ROS 2 did you install and how did you do that?

**Comment by raysu1027** [comment]
Hi alsora,
It's ROS 2 Galactic. I am follw this 
link. https://iroboteducation.github.io/create3_docs/setup/ubuntu2004/

![image](https://user-images.githubusercontent.com/110970753/231995153-84c816d2-09b4-4d39-8097-8c0e98e25215.png)

**Comment by alsora** [comment]
Those instructions are for a desktop (x86_64) build of ROS 2.
I think you are running on a Jetson Xavier, so those instruction are likely not correct.

**Comment by raysu1027** [comment]
Hi alsora,
But create3 documentation can use on jenson.
like this https://iroboteducation.github.io/create3_docs/setup/jetson/
Am i missing something?

**Comment by alsora** [comment, marked as answer]
Yes, but note that these instructions are different from those you linked before.

You should make sure to follow the Jeston-specific instructions here: https://iroboteducation.github.io/create3_docs/setup/jetson/

And FYI, this problem is not related to Create 3 software, but rather "how to setup ROS 2 on NVidia jetson".

**Reply by raysu1027** [reply]
Hi alsora

Thanks for your help.
In this link https://iroboteducation.github.io/create3_docs/setup/jetson/
It's build success.
![image](https://user-images.githubusercontent.com/110970753/232678347-3c87f335-2f97-4d5e-ba4b-63c268daadf3.png)

But have another question.
I want use this case. https://edu.irobot.com/learning-library/teleop-twist-with-the-create-3
**ros-galactic-teleop-twist-keyboard** is already install.
And run command **ros2 run teleop_twist_keyboard teleop_twist_keyboard**
Can not found **teleop_twist_keyboard**
![image](https://user-images.githubusercontent.com/110970753/232679920-ee37c64a-3c49-4991-86ec-5f4ba773204f.png)

**Reply by alsora** [reply]
Can you do
```
source /opt/ros/galactic/setup.sh
source ~/create3_examples_ws/install/setup.sh
```

and then try again
```
ros2 run teleop_twist_keyboard teleop_twist_keyboard 
```


(P.S. don't post screenshots of the terminal, but rather copy-paste it, as it makes the code easier to read/find/copy)

**Reply by raysu1027** [reply]
Hi alsora,

Thank you for your reminder, I will pay attention to the writing format.
And Robot it's work.
But it has two Error message.
```
Error opening '/root' -> Function loadXML
Error parsing '/root' -> Function loadXMLFile
```
Does it have any effect?


================================================================================

```
root@ros2-desktop:~/create3_examples_ws/src/create3_examples/create3_teleop/launch# ros2 run teleop_twist_keyboard teleop_twist_keyboard
2023-04-18 08:04:32.689 [XMLPARSER Error] Error opening '/root' -> Function loadXML
2023-04-18 08:04:32.689 [XMLPARSER Error] Error parsing '/root' -> Function loadXMLFile

This node takes keypresses from the keyboard and publishes them
as Twist messages. It works best with a US keyboard layout.

Moving around:
   u    i    o
   j    k    l
   m    ,    .

For Holonomic mode (strafing), hold down the shift key:

   U    I    O
   J    K    L
   M    <    >

t : up (+z)
b : down (-z)

anything else : stop

q/z : increase/decrease max speeds by 10%
w/x : increase/decrease only linear speed by 10%
e/c : increase/decrease only angular speed by 10%

CTRL-C to quit

currently:	speed 0.5	turn 1.0
```
================================================================================

**Reply by alsora** [reply]
Great to hear that it works now!

About the error messages: 
```
Error opening '/root' -> Function loadXML
Error parsing '/root' -> Function loadXMLFile
```

these indicate that the CycloneDDS XML profile that you are using is likely malformed.
You should have it set it up with 
```
export CYCLONEDDS_URI='<CycloneDDS><Domain><General><NetworkInterfaceAddress>l4tbr0</NetworkInterfaceAddress></General></Domain></CycloneDDS>'
```

I'm not sure if this is a problem or not, but if you see communication issues, please don't hesitate to opena a new discussion here in the forum and we'll try to help you.

**Reply by raysu1027** [reply]
Thank you for your help.

**Comment by shamlian** [comment]
Also, to chime in here: Jetson support is via the community, and this is issue is not with Create 3 examples; I am moving this to a discussion, and we can continue it there with the community.
